nested-data-and-nested-loops/practice-nested-data-and-nested-loops.md

```diff
@@ -0,0 +1,403 @@
+  def test_student_name(self):
+    name = get_student_name()
+    expected = "Lizzie"
+    self.assertEqual(name, expected)
+```
+##### !end-tests
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: b54addb3-48b0-462a-9f6a-f9b749fba39f
+* title: Practice Nested Data IV
+
+##### !question
+Modify `get_students()` to print all names from `students`.
+##### !end-question
+
+##### !placeholder
+```python
+students = [
+  ['Michelle', 'Beyonce', 'Kelly'],
+  ['Luke', 'Leia', 'Han'],
+  ['Gordon', "Lizzie", "Miranda"]
+]
+
+def get_students():
+    pass
+```
+##### !end-placeholder
+##### !tests
+```python
+
+import main
+import unittest
+from unittest.mock import patch 
+
+@patch('builtins.print')
+class TestStudent(unittest.TestCase):
+  def test_students(self, mock_print):
+    get_students()
+    mock_print.assert_called_with('Miranda')
+```
+##### !end-tests
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: 613ccf11-e7e6-4fad-a1b1-6a962bcd8967
+* title: Practice Nested Data V
+
+##### !question
+Modify `get_school_name()` to return `University of Washington` from the `schools` list.
+##### !end-question
+
+##### !placeholder
+```python
+schools = {
+    "school 1": {
+        "name": "Western Washington University",
+        "city": "Bellingham"
+    },
+    "school 2": {
+        "name": "University of Washington",
+        "city": "Seattle"
+    },
+    "school 3": {
+        "name": "Washington State University",
+        "city": "Pullman"
+    }
+}
+
+def get_school_name():
+    pass
+```
+##### !end-placeholder
+##### !tests
+```python
+
+import main
+import unittest
+
+class TestSchool(unittest.TestCase):
+  def test_school_name(self):
+    name = get_school_name()
+    expected = "University of Washington"
+    self.assertEqual(name, expected)
+```
+##### !end-tests
+### !end-challenge
+<!-- prettier-ignore-end -->
+
+<!-- prettier-ignore-start -->
+### !challenge
+* type: code-snippet
+* language: python3.6
+* id: b54addb3-48b0-462a-9f6a-f9b749fba39f
+* title: Practice Nested Data VI
+
+##### !question
+Modify `get_schools` to print the name and city of each school from `schools`.
+##### !end-question
+
+##### !placeholder
+```python
+schools = {
+    "school 1": {
+        "name": "Western Washington University",
+        "city": "Bellingham"
+    },
+    "school 2": {
+        "name": "University of Washington",
+        "city": "Seattle"
+    },
+    "school 3": {
+        "name": "Washington State University",
+        "city": "Pullman"
+    }
+}
+
+def get_schools():
+    pass
+```
+##### !end-placeholder
+##### !tests
+```python
+
+import main
+import unittest
+from unittest.mock import patch 
+
+@patch('builtins.print')
+class TestSchools(unittest.TestCase):
+  def test_schools(self, mock_print):
```

Comment:
Same feedback.  Consider multiple cases (use of `global`) and `StringIO` to capture results with full fidelity.